### DISCUSSION TITLE
nuxt3 feels so naked (sub optimal for newcomers and beginners)

### DISCUSSION THREAD

**Discussion by lustremedia** [nuxt/nuxt#17445] [General]
I just installed a new nuxt3 project for the first time and I was wondering why the welcome component is hidden in the node_module/nuxt3/dist folder.

`if (true) {
    nuxt.app.component("NuxtWelcome", __vite_ssr_import_0__.defineAsyncComponent(() => __vite_ssr_dynamic_import__('/node_modules/nuxt3/dist/app/components/nuxt-welcome.vue')));
  }`

What is the reasoning behind this? 
Imho, It is always nice to have some working code that one can play around with, right away. 
For someone who starts with nuxt3, it is really puzzeling where that inital component comes from.

Is there a way to generate the optional folder structure to get a sort of nuxt2 feeling?

Is the reasoning of nuxt3 to strive for maximum flexibility through max minimalism?

nuxt2 felt really home straight away (sort of like a cozy apartment), nuxt3 is more like, ok, I need to fully decorate the apartment from scratch and where are my clothes.  :-)

**Update:**

I have written to a core member and I was given this information:
_"We plan to have different boilerplate to start with and a better CLI
This is planned but we move step by step"_

So, I guess this will be better once nuxt3 hits production ...

**Comment by ivodolenc** [comment]
It may not be a bad idea to allow users to select the `files`, `folders` and `features` they really want to use. I believe this is the reason why the default `nuxt3` project is currently “naked” 😀 (thumbs up for the team on that decision 👍 )

We can always create our own small private `starter-template` tailored to our needs. That way we can create new projects in seconds.

**Reply by lustremedia** [reply]
Yes, but this does not adress newcomers to the the nuxt3 project. I had such a nice time adjusting to nuxt2 coming from rails because it had an initial overhead and you could see where the screws and bolts are. This is not the case for nuxt3.

**Comment by mirabledictu** [comment]
For me, naked is better so that we can create our own templates or do whatever we want. We are not locked.

**Reply by lustremedia** [reply]
Obiviously you are not a beginner or new to nuxt, right?

**Reply by jolo-dev** [reply]
But still nuxt has a folder convention or can I put my components inside a `custom`- folder?
I totally agree. I was shocked that there no other folders.
There is this `-o`- folder when I first time run the dev server.

**Comment by ken-playerlync** [comment]
@ivodolenc Ok so it was on purpose, that makes sense.. I thought there was a bug with installing it when I did the install myself. That should be in the documentation that it's a la carte, I was expecting the same boilerplate and structure like Nuxt 2. Like @lustremedia described. I was surprised it did not walk me through the options like Nuxt 2....

**Comment by lustremedia** [comment]
nuxt3 is a bit of a paradigm shift. nuxt2 is more like "convention over configuration". Alot of configuration is taken care of from the start. Of course there is still plenty convention, but one has to digg deeper in nuxt3 than with nuxt2, and it is not so intuitve from the start. Especially if you have never touched nuxt before.

In nuxt3 for example, you need to create a page/index.vue in order to get vue router. But on the other hand if you did not need vue router in nuxt2 you still most of the time shipped it anyway, cause the project was started with it.

With nuxt3 I am for example not sure how to config tailwindcss, need to search how it is done and then get it going. Nuxt2 did most of the configuration for me to get the first lift.

It would be nice to have a tad more preconfigured options for popular modules etc. but also it is getting used to new conventions in nuxt 3.

**Reply by zeppytoh** [reply]
By the way I was also having a lot of trouble getting tailwindcss to work. I'm very new to Vue and build tools so had been depending on the documentation from Tailwindcss and nuxt-tailwind. For what it's worth, I just found Windi CSS and they have a module for nuxt3. Can anyone chime in on the differences or pros and cons against Tailwindcss?

**Reply by lustremedia** [reply]
AFAIK windicss and tailwind are very similar and they are somewhat rival util first css frameworks. Tailwind is more popular and better funded.

**Reply by zeppytoh** [reply]
Thank you, that's very helpful. At the moment I'll have to go with Windi CSS since I don't know how to get Tailwind to work with v3.

**Reply by curtgrimes** [reply]
Here is some more historical perspective on the [differences between Windi and Tailwind](https://reddit.com/r/Nuxt/comments/qc4p1s/_/hheyvvn/?context=1). After doing some of my own research and getting this working with Nuxt 3, I now prefer Windi.

**Comment by curtgrimes** [comment]
I think there would be value to beginners in having a very minimal example by way of the `<NuxtWelcome/>` component being included in the components folder after initializing a Nuxt 3 project.

**Reply by dammy001** [reply]
Exactly

**Reply by Twoody** [reply]
Any word on this getting a ticket/put on the roadmap?

**Reply by lustremedia** [reply]
I have not opened a ticket but I would love to see some of the core members to comment on the discussion.

**Comment by wobsoriano** [comment]
Saw this one just now https://github.com/antfu/vitesse-nuxt3

**Comment by amiria703** [comment]
It would be a good idea to have a CLI to configure what is needed, something like what vite has. (`pnpm create vite` command)
